fluent-api/src/main/java/org/codice/imaging/nitf/fluent/NitfParserParsingFlow.java

```diff
@@ -122,11 +134,107 @@ public final NitfSegmentsFlow headerOnly() throws NitfFormatException {
      * @throws NitfFormatException when it's thrown by the parser.
      *
      */
-    public final NitfSegmentsFlow build(final SlottedParseStrategy parseStrategy) throws NitfFormatException {
+    public final NitfParserParsingFlow build(final SlottedParseStrategy parseStrategy) throws NitfFormatException {
```

Comment:
The logic behind switching return types here is that it makes it more obvious what the user of this class should do next.   Basically, you had to call allData(), headerOnly() or build() before you got access to forEachImage (or whatever).  The IDEs auto-complete makes the next step in the workflow obvious.  By merging these two classes together, we lose that and I think it makes the API much harder to use.  I think you can do something like this: https://gist.github.com/dcruver/5a9f28ed0036b58725e994861dd0402c.